app/src/main/java/ua/com/free/localmusic/localmusic/manager/impl/MediaPlayerService.java

```diff
@@ -0,0 +1,207 @@
+package ua.com.free.localmusic.localmusic.manager.impl;
+
+import android.app.Service;
+import android.content.Intent;
+import android.media.AudioManager;
+import android.media.MediaPlayer;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import ua.com.free.localmusic.LocalMusicApplication;
+import ua.com.free.localmusic.api.youtuberipper.YoutubeRipperAPI;
+import ua.com.free.localmusic.events.OnPlayNextEvent;
+import ua.com.free.localmusic.events.OnPlaySongEvent;
+import ua.com.free.localmusic.events.OnSetPlaylistAndPlayEvent;
+import ua.com.free.localmusic.events.OnSetPlaylistEvent;
+import ua.com.free.localmusic.localmusic.manager.IMediaPlayerManager;
+import ua.com.free.localmusic.localmusic.manager.IPlaylist;
+import ua.com.free.localmusic.models.Song;
+import ua.com.free.localmusic.networkoperations.model.SongFromRipperServiceModel;
+
+/**
+ * Created by Dmitry Risovanyi
+ * on 30.08.2017.
+ */
+
+public class MediaPlayerService extends Service implements IMediaPlayerManager, MediaPlayer.OnCompletionListener,
```

Comment:
Actually it's not clear to me, why did you copy-pasted 90% of the code from MediaPlayerManager to this file? In a terms of OOP we need to REuse code and from my point of view class MediaPlayerManager can be easily used in this service. Also I think that it's a bad idea to move player logic inside a raw service. Better to move it in some other class and then call methods of this class from a service